File: sdk-playwrigth/pages/metamask/metametrics.go
package metamask

import (
	"fmt"

	"github.com/playwright-community/playwright-go"
)

type Metametrics struct {
	page playwright.Page
}

func NewMetametrics(page playwright.Page) *Metametrics {
	return &Metametrics{
		page: page,
	}
}

func (m *Metametrics) Disagree() (*Completion, error) {
	err := m.page.Locator(`//*[@id="metametrics-opt-in"]`).Click()
	if err != nil {
		return nil, fmt.Errorf("клик чекбокса: %w", err)
	}

	err = m.page.Locator(`//*[@id="app-content"]/div/div/div/div[2]/div/div[4]/button`).Click()
	if err != nil {
		return nil, fmt.Errorf("клик кнопки: %w", err)
	}

	page := NewCompletion(m.page)

	return page, nil
}

File: tests/ui/pages/metamask/MetaMetrics.ts
import { expect, Page } from "@playwright/test";
import { BasePage } from "../BasePage";
import { MetamaskOnboardingCompletion } from "./OnboardingCompletion";

export class MetamaskMetrics extends BasePage {
  constructor(page: Page) {
    super(page);
    this.page = page;
    this.url = "chrome-extension://idngmeffehjdaahfoiejjnpahdedomlh/home.html#onboarding/metametrics";
  }

  private SELECTORS = {
    USAGE_DATA_CHECKBOX: this.page.locator('//*[@id="metametrics-opt-in"]'),
    CONTINUE_BUTTON: this.page.locator('//*[@id="app-content"]/div/div/div/div[2]/div/div[4]/button')
  }

  async usageDataCheckboxClick() {
    await this.SELECTORS.USAGE_DATA_CHECKBOX.click();
  }

  async continueClick()  {
    await this.SELECTORS.CONTINUE_BUTTON.click();

    var nextPage = new MetamaskOnboardingCompletion(this.page)
    await nextPage.verifyRequiredElementsPresent()

    return nextPage
  }

  async navigateTo() {
    await this.page.goto(this.url);
  }

  async verifyRequiredElementsPresent() {
    var timeout = 10*1000
    await expect(this.SELECTORS.USAGE_DATA_CHECKBOX).toBeVisible({timeout:timeout})
    await expect(this.SELECTORS.CONTINUE_BUTTON).toBeVisible({timeout:timeout})
  }
}
